sklearn/preprocessing/value_dropper.py

```diff
@@ -0,0 +1,216 @@
+# Author : Raghav RV <rvraghav93@gmail.com>
+#
+# Licence : BSD 3 clause
+
+import numpy as np
+
+from ..utils import check_X_y
+from ..utils import check_random_state
+
+__all__ = ["drop_values"]
+
+
+def drop_values(X, y=None, missing_mask=None,
+                missing_values="NaN",
+                missing_fraction=0.1,
+                label_correlation=1.0,
+                n_labels=1,
+                labels=None,
+                missing_mask_only=False,
+                return_labels=False,
+                return_missing_mask=False,
+                copy=False,
+                random_state=None):
+    """Drop values based on a preset strategy.
+
+    Attributes
+    ----------
+
+    X : ndarray like of shape (n_features, n_samples)
+
+        Data, in which the values must be dropped and set to
+        missing_values.
+
+    y : ndarray like of shape (n_samples,), optional
+        Target, in a supervised classification task.
+
+    missing_mask : bool ndarray shape (n_features, n_samples), optional
+        This is used to either denote the missing values that already
+        exist in the data or simply to specify a missing mask to modify
+        inplace.
+
+    missing_values : {"NaN" (or np.nan) | int}, default "NaN"
+        The missing value to use
+
+    label_correlation : float, default 1.0
+        1 (MNAR) - Randomly choose n_targets (or take from given
+        targets) and correlate the missing values with the occurence
+        of those target labels.
+
+        0 (MCAR) - Randomly drop values without correlating the
+        missingness with any target.
+
+        Any value inbetween would constitute a noisy MNAR missingness.
+
+    n_labels : int, optional, default 1
+        The number of labels to pick at random and correlate with
+
+    labels : 1D list/ndarray, optional, default None
+        The list of labels (must match with the labels in y) to
+        correlate with.
+
+        If this is specified n_labels argument is ignored.
+
+    missing_mask_only : bool, default False
+        Whether to modify/return only the missing mask without
+        touching the actual data X.
+
+    return_missing_mask : bool, default False
+        Whether to return the missing mask along with the data (X, y)
+
+    return_labels : bool, default False
+        Whether to return the picked labels
+
+    copy : bool, default False
+        Whether to copy the data (and missing_mask) or work inplace.
+
+    verbose : bool, default False
+        Whether to print the number of correlated/non-correlated missing values
+        that will be introduced.
+
+    random_state : int, optional
+        The seed for the numpy's random number generator.
+    Returns
+    -------
+
+    X, y (, missing_mask, labels) : Tuple
```

Comment:
`missing_mask_only` case is when you return X and y as such without modifying them but reflecting the (additional) value dropping at `missing_mask`. I think I made it a bit more like "handle-weird-usecases" rather than generic. I'll simplify the function soon.